packages/editor/src/components/editor-global-keyboard-shortcuts/index.js

```diff
@@ -114,6 +99,9 @@ class EditorGlobalKeyboardShortcuts extends Component {
 					bindGlobal
 					shortcuts={ {
 						[ rawShortcut.primary( 's' ) ]: this.save,
+						[ rawShortcut.primary( 'z' ) ]: flow( preventDefault, onUndo ),
```

Comment:
By moving this into `bindGlobal`, we're now preventing default behavior for _all_ inputs in the page, not just RichText, right? We'd then rely on all of those inputs to be fully controlled by state values governed by history? I'm not sure I feel particularly confident we can trust that.

Reply:
Why is the title not part of the history?

Reply:
Non-exhaustive set of others:

- Embed placeholder URL input
- Table placeholder columns / rows inputs
- Media placeholders URL input
- Post permalink
- Categories search

(Discovered via `onSubmit=` Find in Files search)

Reply:
Okay, I'm unsure atm then how to prevent default behaviour for RichText only. Restore https://github.com/WordPress/gutenberg/blob/72e51c692108ce52021564352a6d2661466fc231/packages/editor/src/components/rich-text/remove-browser-shortcuts.js?

Reply:
Yeah, for now I think we might just need to have two KeyboardShortcuts for handling these keys, one top-level but not applied to inputs, the other specific to when a RichText is selected.

tl;dr I think that's what we had in the prior revision you're referring to?

Reply:
Right, I just restored your revision :)